feat(web): show face's border on hovered

oezgueremir opened the pull request:
A new PR for a new branch...the other branch had some unrecoverable issues.

For Screenshots, see  #10413

When watching photos and the info tab (detail-panel) is open, all the faces in the photo are marked.

tasks:
- [x] the "mark all faces" function can be de-/activated on the info tab.
- [x] CircleIconButton mark_all_faces with mdiLabelMultiple icon as controller.
- [x] Each person/face has a different, static border color given. This is provided with a little quick'n'dirty solution. A better solution could/would require changes in the DB model...
- [x] The mouseover event will override that functionallity and mark only the faces of this single person. More over, it will change the visualization of the marking rectangle to border=white, filling=person/face color, opacity=0.6
- [x] If mark_all_faces is active, the CircleIconButton show_hidden_people will always be available. So you are able to see even hidden and unassigned faces. Hidden and unassigned faces will be visualized, as if the mouseover event is triggered for them. It can be further evaluated, if these two states have to be controlled individually and presented with unique designs.

oezgueremir committed:
When watching photos and the info tab (detail-panel) is open, all the faces in the photo are marked.

tasks:
- [x] the "mark all faces" function can be de-/activated on the info tab.
- [x] CircleIconButton mark_all_faces with mdiLabelMultiple icon as controller.
- [x] Each person/face has a different, static border color given.
      This is provided with a little quick'n'dirty solution. A better solution could/would require changes in the DB model...
- [x] The mouseover event will override that functionallity and mark only the faces of this single person.
      More over, it will change the visualization of the marking rectangle to border=white, filling=person/face color, opacity=0.6
- [x] If mark_all_faces is active, the CircleIconButton show_hidden_people will always be available.
      So you are able to see even hidden and unassigned faces.
      Hidden and unassigned faces will be visualized, as if the mouseover event is triggered for them.
      It can be further evaluated, if these two states have to be controlled individually and presented with unique designs.

mertalev commented:
I think it'd look cleaner if the boxes were all white without the different colors. It's not like there's usually multiple faces of a single person in an image; it's generally safe to assume each box is a different person.

oezgueremir commented:
Usually and ideally, yes.
But, and this is where this use case comes from, in many group photos in my library (especially pictures of my children), some people have multiple faces assigned to them.
This feature is not meant for every-day usage. This feature is intended to help you find and fix such problems.

waclaw66 commented:
Nice feature, colored rectangles in people list should be deactivated by default though. It would be cleaner to show them when Mark all faces is activated by clicking that button. Filling face background is quite distracting when a light background color is used under white text person name, I would preffer to make colored border only (first 3 faces).

![obrazek](https://github.com/immich-app/immich/assets/15554561/e093a510-2927-4c7d-b451-55674072af7d)

oezgueremir committed:
change requests:
- [x] In people list, don't fill face background, only draw a border around the thumbnail.
- [x] In people list, only draw border, if "mark all faces" option is active

oezgueremir commented:
> Nice feature, colored rectangles in people list should be deactivated by default though. It would be cleaner to show them when Mark all faces is activated by clicking that button. Filling face background is quite distracting when a light background color is used under white text person name, I would preffer to make colored border only (first 3 faces).

Ok, this is if the option is not active:
![grafik](https://github.com/immich-app/immich/assets/5089754/8a3fac3d-6270-4de3-8f4e-e8de24bc8ca3)

After activation the option (to emphesize my use case, I mapped two pairs of different faces on a single person):
![grafik](https://github.com/immich-app/immich/assets/5089754/e255f42d-c7b8-492d-b7f7-cf9ee6f1fda1)



How do you like the mouse-over for the visualization in the actual picture be handled?

Currently the style is,
border=white, filling=person/face color, opacity=0.6
![grafik](https://github.com/immich-app/immich/assets/5089754/735bccd6-8294-4380-8393-f02a4cf5f700)

This style is used, even if the option is not active. The intention is, to make it always easy to spot all markings in the picture.
![grafik](https://github.com/immich-app/immich/assets/5089754/15f93326-a8f8-4f49-ae4f-52fe8968f51a)

This would be the style in the released version, same two faces marked as the screenshot before
![grafik](https://github.com/immich-app/immich/assets/5089754/2094f5b7-650c-46bc-83a7-ade50f092606)


The style could be bound to the same switch, or to another (global) setting, or (my preference) just always use the new style. With no additional switches to consider.

kingvuke commented:
Will this be taken over? What is the plan?

alextran1502 committed:
Merge branch 'main' of github.com:immich-app/immich into feat/mark-all-faces-v2

alextran1502 committed:
cleaning up

alextran1502 committed:
linting

jrasm91 commented:
This looks pretty good, but it's a bit outdated. Can you help rebase it?

Also, I was thinking that we should...
- add the option to  "associate a color to a person" (#15937)
- add the toggle button to show/hide all faces (override the on:hover functionality - this PR)
- update the hover border to be the person's color (future PR)

For now, can you have the color always be white, and we can implement the per-color portion in a separate PR?
